=== modified file 'addons/web/static/src/js/views.js'
--- addons/web/static/src/js/views.js   2012-06-29 10:53:25 +0000
+++ addons/web/static/src/js/views.js   2014-10-17 14:38:04 +0000
@@ -1064,7 +1066,7 @@
                 });
             }
             trads_mutex.exec(function() {
-                return self.view.dataset.write(self.view.datarecord.id, data, { context : { 'lang': code } });
+                return new session.web.DataSet(self, self.view.dataset.model, self.view.dataset.get_context()).write(self.view.datarecord.id, data, { context : { 'lang': code }})
             });
         });
         this.close();


