=== modified file 'sale/sale.py'
--- sale/sale.py	2013-09-10 15:12:35 +0000
+++ sale/sale.py	2014-11-30 17:53:19 +0000
@@ -1257,7 +1257,7 @@
             result['product_uom'] = product_obj.uom_id.id
             if product_obj.uos_id:
                 result['product_uos'] = product_obj.uos_id.id
-                result['product_uos_qty'] = qty * product_obj.uos_coeff
+                result['product_uos_qty'] = qty * (product_obj.uos_coeff or 1.0)
                 uos_category_id = product_obj.uos_id.category_id.id
             else:
                 result['product_uos'] = False
@@ -1271,14 +1271,14 @@
 
         elif uos and not uom: # only happens if uom is False
             result['product_uom'] = product_obj.uom_id and product_obj.uom_id.id
-            result['product_uom_qty'] = qty_uos / product_obj.uos_coeff
+            result['product_uom_qty'] = qty_uos / (product_obj.uos_coeff or 1.0)
             result['th_weight'] = result['product_uom_qty'] * product_obj.weight
         elif uom: # whether uos is set or not
             default_uom = product_obj.uom_id and product_obj.uom_id.id
             q = product_uom_obj._compute_qty(cr, uid, uom, qty, default_uom)
             if product_obj.uos_id:
                 result['product_uos'] = product_obj.uos_id.id
-                result['product_uos_qty'] = qty * product_obj.uos_coeff
+                result['product_uos_qty'] = qty * (product_obj.uos_coeff or 1.0)
             else:
                 result['product_uos'] = False
                 result['product_uos_qty'] = qty
