=== modified file 'mrp_operations/mrp_operations.py'
--- mrp_operations/mrp_operations.py	2012-01-31 13:36:57 +0000
+++ mrp_operations/mrp_operations.py	2014-11-30 18:27:19 +0000
@@ -249,7 +249,9 @@
         wf_service = netsvc.LocalService("workflow")
         for prod in self.browse(cr, uid, ids):
             if prod.workcenter_lines:
-                wf_service.trg_validate(uid, 'mrp.production.workcenter.line', prod.workcenter_lines[0].id, 'button_start_working', cr)
+               #wf_service.trg_validate(uid, 'mrp.production.workcenter.line', prod.workcenter_lines[0].id, 'button_start_working', cr)
+               for workcenter_line in obj.workcenter_lines:
+                   wf_service.trg_validate(uid, 'mrp.production.workcenter.line', workcenter_line.id, 'button_start_working', cr)   	
         return super(mrp_production,self).action_in_production(cr, uid, ids)
     
     def action_cancel(self, cr, uid, ids, context=None):