=== modified file 'purchase_landed_costs/purchase.py'
--- purchase_landed_costs/purchase.py	2014-01-08 21:26:40 +0000
+++ purchase_landed_costs/purchase.py	2014-11-30 18:43:16 +0000
@@ -20,6 +20,8 @@
             help="Landed cost for stock valuation. It will be added to the \
                    price of the supplier price."
         ),
+	'invoice_id': fields.many2one('account.invoice', 'Invoice',
+            domain=[('type','=','in_invoice')]),
         'amount_currency': fields.float(
             'Amount Currency',
             help="The amount expressed in an optional other currency."
@@ -270,6 +272,7 @@
             vals['currency_id'] = order_cost.currency_id.id
             vals['price_type'] = order_cost.price_type
             vals['picking_id'] = pick_id
+            vals['invoice_id'] = order_cost.invoice_id and order_cost.invoice_id.id or False
             self._logger.debug('vals `%s`', vals)
             cost_obj.create(cr, uid, vals, context=context)
 
@@ -288,6 +291,7 @@
                     vals['currency_id'] = order_cost.currency_id.id
                     vals['price_type'] = order_cost.price_type
                     vals['move_line_id'] = line.id
+                    vals['invoice_id'] = order_cost.invoice_id and order_cost.invoice_id.id or False
                     self._logger.debug('vals `%s`', vals)
                     cost_obj.create(cr, uid, vals, context=context)
         self._logger.debug('cost created')

=== modified file 'purchase_landed_costs/purchase_view.xml'
--- purchase_landed_costs/purchase_view.xml	2013-12-18 22:47:59 +0000
+++ purchase_landed_costs/purchase_view.xml	2014-11-30 18:44:04 +0000
@@ -11,6 +11,7 @@
         <tree string="Landing Costs" editable="bottom">
               <field name="product_id" on_change="onchange_product_id(product_id)"/>
               <field name="partner_id"/>
+              <field name="invoice_id"/>
               <field name="amount"/>
               <field name="price_type"/>
               <field name="amount_currency"/>
@@ -28,6 +29,7 @@
         <form string="Landing Costs">
               <field name="product_id" on_change="onchange_product_id(product_id)"/>
               <field name="partner_id"/>
+              <field name="invoice_id"/>
               <field name="amount"/>
               <field name="price_type"/>
               <field name="amount_currency"/>

=== modified file 'purchase_landed_costs/stock.py'
--- purchase_landed_costs/stock.py	2014-01-08 21:26:40 +0000
+++ purchase_landed_costs/stock.py	2014-11-30 18:45:01 +0000
@@ -228,14 +228,14 @@
                 if picking.move_lines:
                     for ml in picking.move_lines:
                         qty_ratio = ml.product_qty / \
-                            new_picking.quantity_total
+                           (new_picking.quantity_total or 1.0) 
                         value_ratio = ml.sub_total / \
-                            new_picking.total_amount
+                           (new_picking.total_amount or 1.0) 
                         price_unit = (
                             ml.landed_cost +
                             qty_ratio * chg_landed_cost_base_quantity +
                             value_ratio * chg_landed_cost_base_value
-                        ) / ml.product_qty
+                        ) / (ml.product_qty or 1.0) 
                         ml_obj.write(
                             cr, uid, [ml.id],
                             {'price_unit': price_unit},

