=== modified file 'purchase/purchase.py'
--- purchase/purchase.py	2014-06-17 09:58:28 +0000
+++ purchase/purchase.py	2015-07-13 15:28:37 +0000
@@ -931,7 +931,8 @@
                 'notes': product.description_purchase,
                 'taxes_id': [(6,0,taxes)],
             }
-            name = seq_obj.get(cr, uid, 'purchase.order') or _('PO: %s') % procurement.name
+            context.update({'force_company': procurement.company_id.id})
+            name = seq_obj.get(cr, uid, 'purchase.order', context) or _('PO: %s') % procurement.name
             po_vals = {
                 'name': name,
                 'origin': procurement.origin,
