=== modified file 'product_lot_sequence/stock_production_lot_sequence.py'
--- product_lot_sequence/stock_production_lot_sequence.py	2012-07-27 08:40:22 +0000
+++ product_lot_sequence/stock_production_lot_sequence.py	2015-11-26 16:49:14 +0000
@@ -35,7 +35,7 @@
     """
     pool = pooler.get_pool(cr.dbname)
 
-    if company_id:
+    if company_id or True:#Si se carga un producto con la compañía a falso desde un xml falla si no entra en el if, por eso añadimos or True.
         #cache system for massive creation of sequences
         if not _sequences_cache.get(company_id):
             _sequences_cache[company_id] = {}
